core/src/main/java/nav/statusplattform/core/openingHours/ServicesUpTimeRenumerator.java

```diff
@@ -0,0 +1,248 @@
+                created times*/
+                sumOfActualUptime += totalOHMinutesBetweenPeriods - excessStartMinutes - excessEndMinutes;
+            }
+
+            /*Services actual uptime total number of minutes between previous record and current record
+             created times*/
+            sumOfExpectedUptime += totalOHMinutesBetweenPeriods - excessStartMinutes - excessEndMinutes;
+
+            /* Prepare for next iteration of loop */
+            previousRecord = currentRecord;
+        }
+
+        //Obtain the number of minutes from the last record
+        RecordEntity lastRecord = records.getLast();
+
+        /* Sum the total number of time in days between the two periods.  Add 1 if the period between
+        last record and end date is under 1 as the number of days is used in the calculation.
+        */
+
+        long totalMinsBetweenLastRecordAndTo = 0;
+
+        ZonedDateTime currentTime = ZonedDateTime.now();
+        int noOfDays = (int) noOfDaysBetweenTwoPeriods(lastRecord.getCreated_at(), to);
+
+
+        //The total of Opening hours start and end time in minutes
+        if (noOfDays > 1) {
+            totalMinsBetweenLastRecordAndTo = ChronoUnit.MINUTES.between(lastRecord.getCreated_at().toLocalTime(), to.toLocalTime()) * noOfDays;
+        }
+
+
+        /*Calculates minutes from the last record start time if its time is greater than the opening hours start time */
+        long minsToLastRecordActualStart = 0;
+
+        if (withinWorkingHours(lastRecord, from, to)) {
+            if (currentTime.toLocalTime().isBefore(to.toLocalTime())) {
+                minsToLastRecordActualStart = ChronoUnit.MINUTES.between(lastRecord.getCreated_at().toLocalTime(), currentTime.toLocalTime());
+            } else {
+                minsToLastRecordActualStart = ChronoUnit.MINUTES.between(lastRecord.getCreated_at().toLocalTime(), to.toLocalTime());
+            }
+        }
+
+
+        //Last record total minutes
+        long expectedUptimeLastRecord = totalMinsBetweenLastRecordAndTo - minsToLastRecordActualStart;
+        //long expectedUptimeLastRecord = minsFromStartToLastRecordActualStart;
+
+        //if the last record is up, its time is included in the calculation
+        long actualUpTimeLastRecord = 0L;
+        if (lastRecord.getStatus() == ServiceStatus.OK) {
+            actualUpTimeLastRecord = expectedUptimeLastRecord;
+        }
+
+        //total expected time
+        long expectedUptimeTotal = sumOfExpectedUptime + expectedUptimeLastRecord;
+
+        //total actual up time
+        long actualUpTimeTotal = sumOfActualUptime + actualUpTimeLastRecord;
+
+        uptimeTotal = new UpTimeTotal(actualUpTimeTotal, expectedUptimeTotal);
+    }
+
+    private long noOfDaysBetweenTwoPeriods(ZonedDateTime periodOne, ZonedDateTime periodTwo) {
+        // Returns the sum of Opening Hours times between the records previous and current times.
+        return (int) (ChronoUnit.DAYS.between(periodOne, periodTwo));
```

Comment:
Why the cast to int? If it's a float, a specific rounding algo should be chosen.